frontend/app/services/raoidc-service.server.ts

```diff
@@ -129,6 +128,13 @@ async function createRaoidcService() {
 
     const idToken: IdToken = session.get('idToken');
 
+    if (session.has('userInfoToken')) {
+      const userInfoToken: UserinfoToken = session.get('userInfoToken');
+      if (userInfoToken.mocked) {
+        return true;
+      }
+    }
```

Comment:
I think you should move the session.has check up a few lines.. and then also log a message when the user is mocked. Like this:

``` typescript
if (!session.has('idToken') || !session.has('userInfoToken')) {
  log.debug(`User has not authenticated; redirecting to /auth/login?returnto=${returnTo}`);
  throw redirect(`/auth/login?returnto=${returnTo}`);
}

const idToken: IdToken = session.get('idToken');
const userInfoToken: UserinfoToken = session.get('userInfoToken');

if (userInfoToken.mocked) {
  log.debug('Mocked user; skipping RAOIDC session validation');
  return true;
}
```

I think it is important to log when we're skipping validation, just in case it might be useful when troubleshooting.

Reply:
done